src/base/bittorrent/session.cpp

```diff
@@ -4647,8 +4655,15 @@ void Session::handleListenFailedAlert(const lt::listen_failed_alert *p)
 
 void Session::handleExternalIPAlert(const lt::external_ip_alert *p)
 {
+    const QHostAddress externalIP {toString(p->external_address)};
     LogMsg(tr("Detected external IP: %1", "e.g. Detected external IP: 1.1.1.1")
-        .arg(toString(p->external_address)), Log::INFO);
+        .arg(externalIP.toString()), Log::INFO);
+    if (isReannounceWhenAddressChanged() && !m_lastExternalIP.isNull() && m_lastExternalIP != externalIP)
+    {
+        for (const lt::torrent_handle &torrent : m_nativeSession->get_torrents())
```

Comment:
extract this loop into a class method.

Reply:
Already moved into `Session::reannounceToAllTrackers` method.